psi4/src/psi4/libfock/jk.cc

```diff
@@ -72,11 +72,12 @@ std::shared_ptr<JK> JK::build_JK(std::shared_ptr<BasisSet> primary, std::shared_
                                  Options& options, std::string jk_type) {
 
     bool is_composite = jk_type.find("+") != std::string::npos; // does SCF_TYPE contain +?
+    is_composite = is_composite || options.get_str("SCF_TYPE") == "DFDIRJ"; // is SCF_TYPE equal to DFDIRJ? 
```

Comment:
Just added a new implementation for `is_composite` that uses `std::any_of` to handle it more elegantly! It will technically evaluate `true` if `SCF_TYPE` is set to `COSX` or `LINK`, but those are never accessible options via `SCF_TYPE` normally anyway. Still worth testing for those cases, too, to make sure the code throws right.